extensions/amp-social-share/0.2/social-share.js

```diff
@@ -30,63 +32,196 @@ const DEFAULT_HEIGHT = 44;
-        throw new Error('no navigator.share');
-      }
-      const dataStr = href.substr(href.indexOf('?'));
-      const data = parseQueryString(dataStr);
-      navigator.share(data).catch(() => {
-        // TODO(alanorozco): Warn here somehow.
-        // warn(TAG, e.message, dataStr);
+    if (bindings) {
+      Object.keys(bindings).forEach((name) => {
+        combinedBindings[name.toUpperCase()] = bindings[name]
+          ? bindings[name]
+          : '';
       });
+    }
+    const finalEndpoint = Object.keys(combinedBindings).reduce(
+      (endpoint, binding) =>
+        endpoint.replace(new RegExp(binding, 'g'), combinedBindings[binding]),
+      endpointWithParams
+    );
+    return finalEndpoint;
+  }
+
+  /**
+   * @private
+   * @param {?string} message
+   */
+  function throwError_(message) {
+    throw new Error(message);
+  }
+
+  /**
+   * @private
+   * @param {?string} message
+   */
+  function throwWarning_(message) {
+    console.warn(message);
+  }
+
+  /**
+   * @private
+   */
+  function handleActivation_() {
+    if (finalEndpoint.split(':', 1)[0] === 'navigator-share') {
+      if (window && window.navigator && window.navigator.share) {
+        const dataStr = finalEndpoint.substr(finalEndpoint.indexOf('?'));
+        const data = parseQueryString(dataStr);
+        window.navigator.share(data).catch((e) => {
+          throwWarning_(`${e.message}. ${name}`);
+        });
+      } else {
+        throwWarning_(
+          `Could not complete system share.  Navigator unavailable. ${name}`
+        );
+      }
     } else {
       const windowFeatures = 'resizable,scrollbars,width=640,height=480';
-      openWindowDialog(window, href, target, windowFeatures);
+      openWindowDialog(window, finalEndpoint, '_blank', windowFeatures);
     }
   }
 
-  const type = props['type'].toUpperCase();
+  /**
+   * @private
+   * @param {!Event} event
+   */
+  function handleKeyPress_(event) {
+    const {key} = event;
+    if (key == Keys.SPACE || key == Keys.ENTER) {
+      event.preventDefault();
+      handleActivation_();
+    }
+  }
+
+  const {type} = props;
   const baseStyle = CSS.BASE_STYLE;
-  const backgroundStyle = CSS[type];
+  //Default to gray (d3d3d3) background if type is not preconfigured
+  const backgroundStyle = CSS[type.toUpperCase()] || {
+    'backgroundColor': 'd3d3d3',
+  };
   const size = {
-    width: props['width'] || DEFAULT_WIDTH,
-    height: props['height'] || DEFAULT_HEIGHT,
+    width: checkedWidth,
+    height: checkedHeight,
   };
-  useResourcesNotify();
   return (
     <div
       role="button"
       tabindex={props['tabIndex'] || '0'}
-      onKeyDown={handleKeyPress}
-      onClick={handleActivation}
+      onKeyDown={handleKeyPress_}
+      onClick={handleActivation_}
       style={{...size, ...props['style']}}
       {...props}
     >
       <SocialShareIcon
         style={{...backgroundStyle, ...baseStyle, ...size}}
-        type={type}
+        type={type.toUpperCase()}
```

Comment:
Let's get the `type.toUpperCase()` value once somewhere up top and use it elsewhere.